R/tweets_and_users.R

```diff
@@ -40,25 +40,13 @@ tweets_with_users <- function(x) {
 #' @rdname tweets_with_users
 #' @export
 users_with_tweets <- function(x) {
-  if (is.null(x) || (length(x) == 0L && !is.data.frame(x))) {
-    x <- data.frame()
-    attr(x, "tweets") <- data.frame()
-    return(x)
-  }
-  tweets <- status_object(x)
-  tweets <- tweets_df_(tweets)
-  users <- user_object(x)
-  users <- users_df_(users)
-  if (nrow(tweets) == nrow(users)) {
-    if (has_name_(users, "user_id")) {
-      tweets$user_id <- users$user_id
-      if (has_name_(users, "screen_name")) {
-        tweets$screen_name <- users$screen_name
-      }
-    }
-  }
-  attr(users, "tweets") <- tweets
-  join_rtweet(users)
+  users <- users_df_(user_object(x))
+  
+  tweets <- tweets_df_(status_object(x))
+  tweets$user_id <- users$user_id
+  tweets$screen_name <- users$screen_name
+  
+  structure(users, tweets = tweets)
```

Comment:
I was a bit lost the way the attribute was added. But why add it? To return the information provided?

Reply:
So you can optionally get the tweet information, if you want it, but it doesn't otherwise clutter up the user info.

Reply:
I've beefed up the docs for `tweets_data()` and `users_data()` so hopefully the reasoning is more clear now.

Reply:
Much better, now I could understood why this is done.